framework/Samplers/Sampler.py

```diff
@@ -711,9 +711,7 @@ def generateInput(self,model,oldInput):
       # DO format the data as atleast_1d so it's consistent in the ExternalModel for users (right?)
       rlz['inputs'] = dict((var,np.atleast_1d(inExisting[var])) for var in self.restartData.getVars('input'))
       rlz['outputs'] = dict((var,np.atleast_1d(inExisting[var])) for var in self.restartData.getVars('output'))
-      rlz['metadata'] = {'prefix':self.inputInfo['prefix'],
-                         'ProbabilityWeight':self.inputInfo['ProbabilityWeight'],
-                        }
+      rlz['metadata'] = copy.deepcopy(self.inputInfo) # TODO need deepcopy only because inputInfo is on self
```

Comment:
why do we need to copy everything here?

Reply:
How do you know what to copy?  This allows each sampler to do as it needs to, which it turns out is required because "prefix" and "probabilityweight" are not generally sufficient.

Reply:
not sure here, but it is ok to me.